File: app/src/main/java/com/example/phl/data/AppDatabase.java
package com.example.phl.data;

import android.content.Context;
import android.os.AsyncTask;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.example.phl.data.sensation.TactileSensation;
import com.example.phl.data.sensation.TactileSensationDao;
import com.example.phl.data.spasticity.Spasticity;
import com.example.phl.data.spasticity.SpasticityDao;

import java.util.Date;
import java.util.List;

@Database(entities = {TactileSensation.class, Spasticity.class}, version = 3)
@TypeConverters({Converters.class})
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase db;

    public abstract TactileSensationDao tactileSensationDao();
    public abstract SpasticityDao spasticityDao();

    public static AppDatabase getInstance(Context context) {
        if (db == null) {
            db = Room.databaseBuilder(context, AppDatabase.class, "tactile_sensation_database")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return db;
    }
}

File: app/src/main/java/com/example/phl/data/AppDatabase.kt
package com.example.phl.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.example.phl.data.ball.BallTestRaw
import com.example.phl.data.ball.BallTestRawDao
import com.example.phl.data.ball.BallTestResult
import com.example.phl.data.ball.BallTestResultDao
import com.example.phl.data.sensation.TactileSensation
import com.example.phl.data.sensation.TactileSensationDao
import com.example.phl.data.spasticity.Spasticity
import com.example.phl.data.spasticity.SpasticityDao
import com.example.phl.data.tilt.TiltTestResult
import com.example.phl.data.tilt.TiltTestResultDao

@Database(
    entities = [TactileSensation::class, Spasticity::class, BallTestRaw::class, BallTestResult::class, TiltTestResult::class],
    version = 7
)
@TypeConverters(Converters::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun tactileSensationDao(): TactileSensationDao
    abstract fun spasticityDao(): SpasticityDao
    abstract fun ballTestRawDao(): BallTestRawDao
    abstract fun ballTestResultDao(): BallTestResultDao
    abstract fun tiltTestResultDao(): TiltTestResultDao

    companion object {
        private var db: AppDatabase?= null
        fun getInstance(context: Context?): AppDatabase {
            if (db == null) {
                db = databaseBuilder(
                    context!!,
                    AppDatabase::class.java,
                    "tactile_sensation_database"
                )
                    .fallbackToDestructiveMigration()
                    .build()
            }
            return db!!
        }
    }
}